### DISCUSSION TITLE
WENO-OF with compressible solvers

### DISCUSSION THREAD

**Discussion by dylewiczk** [WENO-OF/WENOEXT#51] [Q&A]
Hi,

I was wondering if this library can handle high Mach number flows? If so, have there been any tests coupling it with rhoCentralFoam and rhoPimpleFoam? I know the former one has its own TVD scheme. Would using this library be in direct conflict with rhoCentralFoam built-in capabilities? How about rhoPimpleFoam? I do not think it has any shock capturing capabilities.

Thanks!

**Comment by JanGaertner** [comment]
Hello,

the WENO scheme is implemented so that you can easily use it for any Gauss divergence scheme operator in OpenFOAM. For any solver, any transport equation you solve you can select in the `system/fvSchemes` in the `div` dictionary the WENO scheme. This allows for great flexibility and is familiar with other OpenFOAM schemes, such as TVD vanLeer, limitedLinear, etc. 

You can certainly use it with the rhoPimpleFoam solver to discretize the momentum divergence transport. How well you can capture shocks then depends on not only the discretization of this one scheme but also the general equation system solved. For rhoPimpleFoam specifically, transonic flow systems can be captured. High Mach number flows are probably difficult. 

In contrast to the pressure-based solver, such as rhoPimpleFoam, the density-based solver, rhoCentralFoam, uses a different approach. Here you cannot easily replace the discretization of one divergence term as it is built up completely different from the default pressure-based system of OpenFOAM. 

If you want to simulate high Mach number flows I would advise using a density-based system which is by design well suited for such kind of flows.

Best,

Jan
